packages/test/test-end-to-end-tests/src/test/gc/gcTrailingOps.spec.ts

```diff
@@ -17,200 +17,230 @@ import {
+			key: string,
+			makeReferenced: boolean,
+		) {
+			if (makeReferenced) {
+				fromDataStore._root.set(key, dataStoreToUpdate.handle);
+			} else {
+				fromDataStore._root.delete(key);
+			}
+		}
 
 		/**
-		 * Submits a summary and returns the unreferenced timestamp for all the nodes in the container. If a node is
-		 * referenced, the unreferenced timestamp is undefined.
-		 * @returns a map of nodeId to its unreferenced timestamp.
+		 * Validates that the data store is not deleted from the data store and GC trees in the summary.
+		 * Also, it is referenced / unreferenced as expected.
 		 */
-		async function getUnreferencedTimestamps(summaryTree: ISummaryTree) {
+		function validateDataStoreStateInSummary(
+			summaryTree: ISummaryTree,
+			dataStoreNodePath: string,
+			expectGCStateHandle: boolean,
+			referenced: boolean,
+		) {
+			// Validate that the data store is not deleted from the data store summary tree.
+			const dataStoreId = dataStoreNodePath.split("/")[1];
+			const channelsTree = (summaryTree.tree[channelsTreeName] as ISummaryTree).tree;
+			assert.equal(
+				Object.keys(channelsTree).includes(dataStoreId),
+				true,
+				`Data store ${dataStoreId} should not have been deleted from the summary`,
+			);
+
+			// If expecting the GC state to be a handle, validate that and return.
+			if (expectGCStateHandle) {
+				assert.equal(
+					summaryTree.tree[gcTreeKey].type,
+					SummaryType.Handle,
+					"Expecting the GC tree to be handle",
+				);
+				return;
+			}
+
+			// Validate that the GC state does contains an entry for the data store.
 			const gcState = getGCStateFromSummary(summaryTree);
 			assert(gcState !== undefined, "GC tree is not available in the summary");
-			const nodeTimestamps: Map<string, number | undefined> = new Map();
-			for (const [nodeId, nodeData] of Object.entries(gcState.gcNodes)) {
-				nodeTimestamps.set(nodeId.slice(1), nodeData.unreferencedTimestampMs);
-			}
-			return nodeTimestamps;
+			const dataStoreGCData = gcState.gcNodes[dataStoreNodePath];
+			assert(
+				dataStoreGCData !== undefined,
+				`Data store ${dataStoreNodePath} should be present in GC state`,
+			);
+			assert.equal(
+				dataStoreGCData.unreferencedTimestampMs ? false : true,
+				referenced,
+				`Data store ${dataStoreNodePath}'s referenced state is incorrect`,
+			);
+
+			// Validate that the deleted nodes in the GC data does not have the data store.
+			const deletedNodesState = getGCDeletedStateFromSummary(summaryTree);
+			assert.equal(
+				deletedNodesState?.includes(dataStoreNodePath) ?? false,
+				false,
+				`Data store ${dataStoreNodePath} should not be in deleted nodes`,
+			);
 		}
 
 		beforeEach(async function () {
 			provider = getTestObjectProvider({ syncSummarizer: true });
+			if (provider.driver.type !== "local") {
+				this.skip();
+			}
+			settings["Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"] = sweepTimeoutMs;
+
+			const configProvider = mockConfigProvider(settings);
```

Comment:
nit: This could move up to the top-level scope and just happen once.

Reply:
That would work if the settings doesn't change throughout the tests. However, if it does change from test to test, this doesn't work. That was my assumption in some of the tests and it was wrong. To be consistent, the settings has to be reset updated after every test and the config provider re-created before every test.

Reply:
`settings` is reset in `afterEach`, so I thought tying `configProvider` to that `settings` object could just be done once.  I didn't think deeply about it so you could be right.  It's obv fine as-is

Reply:
I have added this as a follow-up anyway so we can look at it then.